tests/test_statistics_distributions.py

```diff
@@ -431,3 +433,48 @@ def test_methods(self):
 
         assert list(d1.pmf([1, 2])) == [0.00013778100000000018, 0.0014467005000000015]
         assert list(d1.cdf([1, 2])) == [0.00014368590000000018, 0.0015903864000000017]
+
+
+class TestMarcenkoPastur:
+
+    def test_refactoring(self):
+        X = np.array([1.91315248, 1.81189185, 1.72542052, 1.65689567, 1.64134677,
+                      1.62151478, 1.58697418, 1.53606406, 1.51293438, 1.47587683,
+                      1.47113945, 1.43413408, 1.42520198, 1.40412494, 1.3886599,
+                      1.37391991, 1.35689981, 1.34236907, 1.31795082, 1.30957462,
+                      1.29085016, 1.27863071, 1.26960764, 1.25244721, 1.24404906,
+                      1.23392868, 1.21928507, 1.20603135, 1.19923494, 1.1825206,
+                      1.16721998, 1.16247636, 1.13830932, 1.13227976, 1.11265137,
+                      1.10132257, 1.09624903, 1.07900696, 1.06956545, 1.06595237,
+                      1.04785689, 1.04412674, 1.02471149, 1.02143057, 1.01571143,
+                      0.993282, 0.98717485, 0.97478547, 0.96958049, 0.9559686,
+                      0.95172736, 0.94866285, 0.92993663, 0.92236546, 0.91347391,
+                      0.90025847, 0.88726593, 0.88652546, 0.87130921, 0.86891738,
+                      0.85601713, 0.84471017, 0.84039224, 0.82083072, 0.81597161,
+                      0.80310027, 0.79844503, 0.78560994, 0.78167253, 0.77388754,
+                      0.76554336, 0.7586511, 0.7545136, 0.7437965, 0.7401452,
+                      0.73424185, 0.71591487, 0.70597488, 0.69454806, 0.69074335,
+                      0.68826198, 0.66965944, 0.65086124, 0.64712531, 0.6396971,
+                      0.62836684, 0.62156755, 0.61095141, 0.60050098, 0.57893671,
+                      0.57559399, 0.56958534, 0.56116821, 0.54794375, 0.53189713,
+                      0.52770162, 0.51966797, 0.51144757, 0.49444351, 0.47717652])
+        X2 = np.copy(X)
+        n_features0 = 100
+        n_obs0 = 10
+
+        _, sigma1, _ = statistics.marcenko_pastur_fit_params(n_features=n_features0,
+                                                             n_obs=n_obs0,
+                                                             sigma_ini=np.sqrt(0.5),
+                                                             e_val=np.diag(X),
+                                                             bwidth=0.01)
+
+        print(sigma1)
+
+        mp_dist = dis.MarcenkoPastur(0.5, 0.1)
+        #X2 = np.diag(X2)
+        print(X2.shape)
+        # Should n_pts be the same as n_features??
+        mp_dist.fit(X2, n_pts=100, bandwidth=0.01, initial_params=[np.sqrt(0.5), 0.1])
+
+        print(mp_dist.sigma)
+        print(mp_dist.lambda_)
```

Comment:
why print statements here in the test? Shouldn't they be assert statements testing if outputs as desired?